pkg/operator/staticpod/controller/missingstaticpod/missing_static_pod_controller.go

```diff
@@ -0,0 +1,263 @@
+	eventRecorder events.Recorder,
+	targetNamespace string,
+	staticPodName string,
+) factory.Controller {
+	c := &missingStaticPodController{
+		operatorClient:                    operatorClient,
+		podLister:                         kubeInformersForTargetNamespace.Core().V1().Pods().Lister(),
+		configMapListerForTargetNamespace: kubeInformersForTargetNamespace.Core().V1().ConfigMaps().Lister().ConfigMaps(targetNamespace),
+		targetNamespace:                   targetNamespace,
+		operandName:                       staticPodName,
+	}
+	return factory.New().
+		ResyncEvery(time.Minute).
+		WithInformers(
+			operatorClient.Informer(),
+			kubeInformersForTargetNamespace.Core().V1().Pods().Informer(),
+		).
+		WithSync(c.sync).
+		WithSyncDegradedOnError(operatorClient).
+		ToController("MissingStaticPodController", eventRecorder)
+}
+
+func (c *missingStaticPodController) sync(ctx context.Context, syncCtx factory.SyncContext) error {
+	_, originalOperatorStatus, _, err := c.operatorClient.GetStaticPodOperatorState()
+	if err != nil {
+		return err
+	}
+
+	installerPods, err := c.podLister.List(labels.SelectorFromSet(labels.Set{"app": "installer"}))
+	if err != nil {
+		return err
+	}
+
+	// get the most recent installer pod for each node
+	latestInstallerPodsByNode, err := getMostRecentInstallerPodByNode(installerPods)
+	if err != nil {
+		return err
+	}
+
+	errors := make([]string, 0, len(latestInstallerPodsByNode))
+	for node, latestInstallerPodOnNode := range latestInstallerPodsByNode {
+		installerPodRevision, err := getInstallerPodRevision(latestInstallerPodOnNode)
+		if err != nil {
+			// we expect every installer pod to have a installerPodRevision in its name, unexpected error here
+			return fmt.Errorf("failed to get installerPodRevision for installer pod %q - %w", latestInstallerPodOnNode.Name, err)
+		}
+
+		finishedAt, ok := installerPodFinishedAt(latestInstallerPodOnNode)
+		if !ok {
+			// either it's in the process of running, or it ran into an error
+			continue
+		}
+
+		staticPodRevisionOnThisNode := getStaticPodCurrentRevision(node, originalOperatorStatus)
+		gracePeriod, err := c.getStaticPodTerminationGracePeriodSecondsForRevision(staticPodRevisionOnThisNode)
+		if err != nil {
+			return err
+		}
+		threshold := time.Duration(gracePeriod) + 10*time.Second
+
+		if time.Since(finishedAt) > threshold &&
+			staticPodRevisionOnThisNode < installerPodRevision {
+			// if we are here:
+			//  a: the latest installer pod successfully completed at finishedAt
+			//  b: it has been more than 'terminationGracePeriodSeconds' + 10s since the
+			//     installer pod has completed at finishedAt
+			//  c. the static pod is not at correct installerPodRevision yet
+			// then all of the above conditions are true, so we should go degraded
+			syncCtx.Recorder().Eventf("MissingStaticPod", "static pod lifecycle failure - installer pod: %q with revision: %d completed at: %s, static pod revision: %d",
```

Comment:
I think it can produce many events, at least per every `sync` loop. I think we should find a way to avoid it.

Reply:
at maximum the controller will be responsible for 1620 active events with 9 created per minute:
```
3 (kas, etcd, kcm) * 3 (replicas) * 180 (event ttl)
```
event ttl: https://github.com/openshift/cluster-kube-apiserver-operator/blob/69a9edba92634e68[…]57c77aa063d889fdf30bed/bindata/assets/config/defaultconfig.yaml

Reply:
on an sno cluster, in case of a fallback (currentRevision < mostRecentRevision), it can produce events forever.

Reply:
that statement should be valid for any type of cluster not only sno no? Also, even if events will be produced indefinitely, their retention will be very short so it should have a limited impact on the clusters